src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java

```diff
@@ -52,27 +52,39 @@ public void testSubmissionLinkMap() {
         Biomaterial cellSuspension = new Biomaterial("cellSuspension");
 
         Process process1 = new Process("donor-specimen-process");
-        Process process2 = new Process("cellSuspension-file-process");
+        Process process2 = new Process("cellSuspension-sequenceFile-process");
+        Process process3 = new Process("sequenceFile-analysisFile-process");
 
         Protocol collectionProtocol = new Protocol("collectionProtocol");
         Protocol sequencingProtocol = new Protocol("sequencingProtocol");
+        Protocol analysisProtocol = new Protocol("analysisProtocol");
 
-        File file = new File("file");
+        File sequencingFile = new File("sequenceFile");
+        File analysisFile = new File("analysisFile");
 
         specimen.addAsDerivedByProcess(process1);
-        file.addAsDerivedByProcess(process2);
+        sequencingFile.addAsDerivedByProcess(process2);
+        analysisFile.addAsDerivedByProcess(process3);
 
         donor.addAsInputToProcess(process1);
         cellSuspension.addAsInputToProcess(process2);
+        sequencingFile.addAsInputToProcess(process3);
 
         process1.addProtocol(collectionProtocol);
         process2.addProtocol(sequencingProtocol);
+        process3.addProtocol(analysisProtocol);
 
-        when(processRepository.findBySubmissionEnvelope(submissionEnvelope)).thenReturn(Stream.of(process1, process2));
+        when(processRepository.findBySubmissionEnvelope(submissionEnvelope)).thenReturn(Stream.of(process1, process2, process3));
```

Comment:
Why does this test need to be a `SpringBootTest`?
Could it be only a `DataMongoTest`?
This way you would just save everything in the repositories, and there would not be a need to use the `when().thenReturn()`

Reply:
I've converted them as Spring's integration test and just autowired the repository objects as I'm using them in the SubmissionLinkMapController.  It did avoid mocking the repository methods in the tests.